frontend/packages/pipelines-plugin/src/components/catalog/providers/useTasksProvider.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import { TFunction } from 'i18next';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { CatalogItem, ExtensionHook } from '@console/dynamic-plugin-sdk';
+import { getImageForIconClass } from '@console/internal/components/catalog/catalog-item-icon';
+import { useK8sWatchResources } from '@console/internal/components/utils/k8s-watch-hook';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { TaskModel, ClusterTaskModel } from '../../../models';
+import { TaskKind } from '../../../types';
+import { TektonTaskAnnotation, TektonTaskLabel } from '../../pipelines/const';
+
+const normalizeTektonTasks = (TektonTasks: TaskKind[], t: TFunction): CatalogItem<TaskKind>[] => {
+  const normalizedTektonTasks: CatalogItem<TaskKind>[] = _.reduce(
+    TektonTasks,
+    (acc, task) => {
+      const { uid, name, annotations = {}, creationTimestamp, labels = {} } = task.metadata;
```

Comment:
This will throw error if task is undefined/null
```suggestion
      const { uid, name, annotations = {}, creationTimestamp, labels = {} } = task?.metadata ?? {};
```

Reply:
Not sure metadata can be undefined. And I do not believe we can have null tasks because of the origin is from a request of objects.